app/API/V1/Controllers/AuthController.php

```diff
@@ -2,64 +2,174 @@
 	}
-	
+
+    /**
+     * Refresh Tocken
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
 	public function refresh()
 	{
-		if($this->auth->getToken() === FALSE)
-		{
-			throw new BadRequestHttpException('token_absent');
+		if($this->auth->getToken() === FALSE) {
+			throw new BadRequestHttpException(trans('auth.token_absent'));
 		}
 		
-		try
-		{
+		try {
 			$token = $this->auth->refresh();
-		} catch(TokenInvalidException $e)
-		{
-			throw new UnauthorizedHttpException('token_invalid');
+		} catch(TokenInvalidException $e) {
+			throw new UnauthorizedHttpException(trans('auth.token_invalid'));
 		}
 		
 		return response()->json(compact('token'));
 	}
 
+    /**
+     * @param string $provider
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function getSocialAuth(string $provider)
+    {
+        if(!config("services.$provider"))
+            throw new BadRequestHttpException(trans('auth.no_such_provider'));
+
+        /** @var \Laravel\Socialite\Two\FacebookProvider $redirect */
+        $redirect = $this->socialite->driver($provider)->stateless();
+        $result = [
+            'provider' => $provider,
+            'link' => $redirect->redirect()->getTargetUrl()
+        ];
+
+        return response()->json(compact('result'));
+    }
+
+    /**
+     * @param string $provider
+     * @return \Illuminate\Http\JsonResponse
+     * @throws \Doctrine\DBAL\ConnectionException
+     */
+    public function getSocialAuthCallback(string $provider)
+    {
+        $token = null;
+        if($socializeUser = $this->socialite->driver($provider)->stateless()->user()) {
+            $user = $this->userRepo->findOneBy(['email' => $socializeUser->email]);
+            if (!$user) {
+                /** Generate the user based on the socialite data **/
+                $this->userRepo->registerSocializeUser($provider, $socializeUser);
+                $user = $this->userRepo->findOneBy(['email' => $socializeUser->email]);
+                /** Set user's default role **/
+                $this->roleRepo->setUserPermissions($user);
+            }
+
+            $credentials = [
+                'email' => $user->getEmail(),
+                'password' => Hash::make($user->getSocialize()->getToken())
+            ];
+            if ($user) {
+                try {
+                    if(($token = $this->auth->attempt($credentials)) === FALSE) {
+                        return response()->json(['error' => trans('auth.invalid_credentials')], 401);
+                    }
+                } catch(JWTException $e) {
+                    return response()->json(['error' => trans('auth.could_not_create_token')], 500);
+                }
+            }
+        } else {
+            throw new BadRequestHttpException(trans('auth.something_went_wrong'));
+        }
+
+        return response()->json(compact('token'));
+    }
+
+    public function forgotPassword(Request $request)
+    {
+        $token = null;
```

Comment:
Is this just a start and we will put more functionality in?

Reply:
Yes, that is just a starting point for forgot password algo